function-components-todo/src/App.js

```diff
@@ -0,0 +1,97 @@
+import './App.css';
+import { useState,useCallback, useMemo } from "react";
+import uuid from 'react-uuid';
+import ArrowDown from "./Components/Button";
+import InputContainer from "./Components/Input";
+import TodoItem from "./Components/TodoItem";
+import Footer from "./Components/Footer";
+import Box from '@mui/material/Box';
+import { styled } from '@mui/material/styles';
+
+const Top=styled(Box)`
+  border: 1px solid grey;
+  width:600px;
+  height:60px;
+  display: flex;
+  margin: auto;
+`;
+
+function App() {
+  //initial state
+   const [all,setAll]=useState([]);
+   const [newToDo,setNewToDo]=useState("");
+   const [mode,setMode]=useState("all");
+   const [editToDo,setEditToDo]=useState("");
+
+   //adding todo
+   const addTodo=useCallback(()=>{setAll([...all,{id:uuid(),action:newToDo,done:false,isEdit:false}]);
+            setNewToDo("");},[all,newToDo]);
+
+   //count of all list
+   const All=useMemo(()=>[...all].length,[all]);
```

Comment:
variable names and function names should be "all", "completed" and so on. 
Only the components should start with Uppercase char.